### PR TITLE
Adding more detailed counters for texture memory consumption

### PR DESCRIPTION
**Pull request by samcake**
Introducing 2 new counters visible on the stats ui:
- in the gpu texture list the Framebuffer which should account fo r the amount of memory allocated on gpu for textures which are used in Framebuffers.

- added a counter for measuring the estimated cost of all the swapchains coming from the window providing the rendering surface to a gl::Context. The cost of the swapchain depends on its pixel format and the size of the window.

I choose to NOT take into account the swapchain length (double buffered vs single bufferd) to evaluete the pixel cost of the swapchain since in practice the double buffer becomes really something managed by the OS and we can't really know if its really allocated for sure.
Our experimentation and comaprison with the metric comming from Process Explorer seems to validate that...

### CODE REVIEW

File: libraries/gl/src/gl/Context.cpp [hunk 1]
```diff
@@ -40,6 +40,27 @@ static bool enableDebugLogger = QProcessEnvironment::systemEnvironment().contain
 
 using namespace gl;
 
+
+std::atomic<size_t> Context::_defaultFBOMemoryUsage { 0 };
+
+size_t Context::getDefaultFBOMemoryUsage() { return _defaultFBOMemoryUsage.load(); }
+
+size_t Context::evalMemoryUsage(uint32_t width, uint32_t height) {
+    return width * height * 4;
```
Comment (ZappoMan) [inline libraries/gl/src/gl/Context.cpp]: magic number? can you make this a constant with some kind of a description of what it is... why 4?

File: libraries/gpu-gl/src/gpu/gl/GLTexture.cpp [hunk 2]
```diff
@@ -181,8 +181,11 @@ GLTexture::~GLTexture() {
             // the GL45Texture destructor for doing any required work tracking GPU stats
             backend->releaseTexture(_id, _size);
         }
+
+        if (!_external && !_transferrable) {
+            Backend::updateTextureGPUFramebufferMemoryUsage(_size, 0);
+        }
     }
-    Backend::updateTextureGPUVirtualMemoryUsage(_virtualSize, 0);
```
Comment (ZappoMan) [inline libraries/gpu-gl/src/gpu/gl/GLTexture.cpp]: can you explain this change?

Comment (ZappoMan) [inline libraries/gpu-gl/src/gpu/gl/GLTexture.cpp]: oh wait... I see you've removed *VirtualMemoryUsage*

**Comment by ZappoMan** [comment]
Test Plan:
- no change in application behavior
- new stats should be visible

File: libraries/gl/src/gl/Context.cpp (hunk 1 above)
Comment (ZappoMan) [inline libraries/gl/src/gl/Context.cpp]: is it size of RGBA pixel??

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b2b68e69186129111b736b18751724dce906ef0 available at: https://gist.github.com/57e481c923e58d8df272e879cf09f9a7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8901-6b2b68e.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8901-6b2b68e.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 507d3e5a39605c620d4b9d7aa2c1f5e9ed85c656 available at: https://gist.github.com/57e481c923e58d8df272e879cf09f9a7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8901-507d3e5.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8901-507d3e5.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c586bef45216c0379929b7dedab531e57e1ed5a9 available at: https://gist.github.com/57e481c923e58d8df272e879cf09f9a7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8901-c586bef.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8901-c586bef.dmg

**Comment by samcake** [comment]
Here are some first results:
`desktop or HMD	D	D	D	D	D
Desktop Window resolution: 
[width K] x [height K]	1 x 0.5
quarter of a hd screen	1 x 1
horizontal half of a hd screen	1x 2
hd screen full	2x2
horizontal half of 4K screen	4x2
4K screen full
gpu::Texture Commited [MB]	105	143	198	311	571
gl swapchain [MB]	26	60	121	240	482
gml textures  [MB]	8	19	39	79	159
total Interface  [MB]	139	222	358	630	1212
Process Explorer: GPU Dedicated Bytes  [MB]	196	222	372	570	1021
`

**Comment by hifi-gustavo** [comment]
Build console output for commit: 133b0b38fdd20d140eab6e10eb2024649dbd679b available at: https://gist.github.com/57e481c923e58d8df272e879cf09f9a7<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8901-133b0b3.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8901-133b0b3.dmg

**Pull request merged**